torch/csrc/jit/passes/utils/memory_dag.cpp

```diff
@@ -1,11 +1,31 @@
 #include "memory_dag.h"
 
+#include <c10/util/flat_hash_map.h>
 #include <torch/csrc/utils/memory.h>
 #include <algorithm>
 #include <queue>
 
 namespace torch {
 namespace jit {
+namespace {
+std::unordered_set<const Element*, unsigned> comprMap;
```

Comment:
Can you just add this as a member to Element?

Reply:
I suppose, but we'd need to make it static and expose it outside this module (which we might not want to do).

Reply:
Sorry, I didn't mean add the map.  I meant add an `unsigned` member to each Element instance, so that you don't need this map at all.

Reply:
ah, we could if we added a static `index` variable to the constructor of `Element`.  We'd still need the `decomprMap`, but that does remove this.

Reply:
Yeah.  Paired with my suggestion to change `decomprMap` to a vector, I think we could achieve having no hash tables in here. :-)